### ISSUE TITLE
Regression Bug: InitOpenCL() crashes on 17.4.1

### ISSUE THREAD

**Issue by Xaymar** [GPUOpen-LibrariesAndSDKs/AMF#79]
Initially thought to be a coding error, but it actually turned out to crash even in the sample applications. Exact reason unknown, but have reproduced this across two computers now.

### Exception Message:
> Unhandled exception at 0x000001FE002B0560 in obs64.exe: 0xC0000005: Access violation executing location 0x000001FE002B0560.

### Call Stack:
```
 	000001fe002b0560()	Unknown
 	amdocl12cl64.dll!00007ffe0fc2ea27()	Unknown
 	amdocl12cl64.dll!00007ffe0fc30b17()	Unknown
 	amdocl12cl64.dll!00007ffe0fc2f095()	Unknown
 	amdocl12cl64.dll!00007ffe0fc48746()	Unknown
 	amdocl64.dll!00007ffe11eb7df9()	Unknown
 	amdocl64.dll!00007ffe118e81bf()	Unknown
 	amdocl64.dll!00007ffe118c7b8b()	Unknown
 	amfrt64.dll!00007ffe1c2214c5()	Unknown
 	amfrt64.dll!00007ffe1c21efb2()	Unknown
 	amfrt64.dll!00007ffe1c22311b()	Unknown
 	amfrt64.dll!00007ffe1c1fd9e6()	Unknown
 	amfrt64.dll!00007ffe1c1fd362()	Unknown
 	amfrt64.dll!00007ffe1c1fccb0()	Unknown
 	amfrt64.dll!00007ffe1c1da737()	Unknown
	enc-amf.dll!Plugin::AMD::Encoder::Encoder(Plugin::AMD::Codec codec, std::shared_ptr<Plugin::API::IAPI> videoAPI, Plugin::API::Adapter videoAdapter, bool useOpenCLSubmission, bool useOpenCLConversion, Plugin::AMD::ColorFormat colorFormat, Plugin::AMD::ColorSpace colorSpace, bool fullRangeColor, bool useAsyncQueue, unsigned __int64 asyncQueueSize) Line 135	C++
 	enc-amf.dll!Plugin::AMD::EncoderH265::EncoderH265(std::shared_ptr<Plugin::API::IAPI> videoAPI, Plugin::API::Adapter videoAdapter, bool useOpenCLSubmission, bool useOpenCLConversion, Plugin::AMD::ColorFormat colorFormat, Plugin::AMD::ColorSpace colorSpace, bool fullRangeColor, bool useAsyncQueue, unsigned __int64 asyncQueueSize) Line 50	C++
 	enc-amf.dll!std::make_unique<Plugin::AMD::EncoderH265,std::shared_ptr<Plugin::API::IAPI> & __ptr64,Plugin::API::Adapter & __ptr64,bool,bool,enum Plugin::AMD::ColorFormat & __ptr64,enum Plugin::AMD::ColorSpace & __ptr64,bool,bool,unsigned __int64>(std::shared_ptr<Plugin::API::IAPI> & <_Args_0>, Plugin::API::Adapter & <_Args_1>, bool && <_Args_2>, bool && <_Args_3>, Plugin::AMD::ColorFormat & <_Args_4>, Plugin::AMD::ColorSpace & <_Args_5>, bool && <_Args_6>, bool && <_Args_7>, unsigned __int64 && <_Args_8>) Line 1630	C++
 	enc-amf.dll!Plugin::Interface::H265Interface::H265Interface(obs_data * data, obs_encoder * encoder) Line 894	C++
 	enc-amf.dll!Plugin::Interface::H265Interface::create(obs_data * data, obs_encoder * encoder) Line 832	C++
 	obs.dll!obs_encoder_initialize_internal(obs_encoder * encoder) Line 396	C
 	obs.dll!obs_encoder_initialize(obs_encoder * encoder) Line 415	C
 	obs.dll!obs_output_initialize_encoders(obs_output * output, unsigned int flags) Line 1684	C
 	obs-ffmpeg.dll!ffmpeg_mux_start(void * data) Line 279	C
 	obs.dll!obs_output_actual_start(obs_output * output) Line 233	C
 	obs.dll!obs_output_start(obs_output * output) Line 264	C
 	obs64.exe!AdvancedOutput::StartRecording() Line 1435	C++
 	obs64.exe!OBSBasic::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 750	C++
 	obs64.exe!OBSBasic::qt_metacall(QMetaObject::Call _c, int _id, void * * _a) Line 853	C++
 	Qt5Core.dll!0000000076e41116()	Unknown
 	Qt5Widgets.dll!00000000504e86ac()	Unknown
 	Qt5Widgets.dll!00000000504e85d3()	Unknown
 	Qt5Widgets.dll!00000000504e93f8()	Unknown
 	Qt5Widgets.dll!000000005043139c()	Unknown
 	Qt5Widgets.dll!0000000050407ca0()	Unknown
 	Qt5Widgets.dll!0000000050405913()	Unknown
 	Qt5Core.dll!0000000076e223c9()	Unknown
 	Qt5Widgets.dll!0000000050408fd1()	Unknown
 	Qt5Widgets.dll!0000000050459ecd()	Unknown
 	Qt5Widgets.dll!0000000050457dc3()	Unknown
 	Qt5Widgets.dll!0000000050407ca0()	Unknown
 	Qt5Widgets.dll!0000000050406bfa()	Unknown
 	Qt5Core.dll!0000000076e223c9()	Unknown
 	Qt5Gui.dll!00007ffe2f811c2c()	Unknown
 	Qt5Gui.dll!00007ffe2f814140()	Unknown
 	Qt5Gui.dll!00007ffe2f7f844c()	Unknown
 	Qt5Core.dll!0000000076e67b25()	Unknown
 	user32.dll!00007ffe60431c24()	Unknown
 	user32.dll!00007ffe6043156c()	Unknown
 	Qt5Core.dll!0000000076e67299()	Unknown
 	qwindows.dll!00007ffe3fcd6f59()	Unknown
 	Qt5Core.dll!0000000076e1eda1()	Unknown
 	Qt5Core.dll!0000000076e21237()	Unknown
 	obs64.exe!run_program(std::basic_fstream<char,std::char_traits<char> > & logFile, int argc, char * * argv) Line 1295	C++
 	obs64.exe!main(int argc, char * * argv) Line 1827	C++
 	obs64.exe!WinMain()	C++
 	obs64.exe!__tmainCRTStartup() Line 618	C
 	kernel32.dll!00007ffe606a8364()	Unknown
 	ntdll.dll!00007ffe609d70d1()	Unknown
```

### Files:
* [Large Dump](http://cdn.xaymar.com/private/2017/04/05/07/amf-issue-79.7z) (might need VS2015)

**Comment by MikhailAMD** [comment]
How to reproduce with sample app?

**Comment by Xaymar** [comment]
`VCEEncoderD3D.exe -RENDER OpenCLDX11 -FRAMES 500 -OUTPUT ../deleteme.h264 -CODEC AVC -WIDTH 1920 -HEIGHT 1080 -ADAPTERID 0`

This will immediately crash the VCEEncoderD3D sample. I don't think OpenCLDX11 is even needed, as all my VCEEncoderD3D using tests all fail with a crash.

**Comment by MikhailAMD** [comment]
I tried and do not see the crash. Is it possible that you have NV card installed or remains of the NV driver are present?

**Comment by MikhailAMD** [comment]
Try to uninstall all drivers before installing this one.

**Comment by Xaymar** [comment]
I do not have an Nvidia Card to even install an Nvidia driver. Was the public driver package perhaps reuploaded and the one I have downloaded is faulty?

**Comment by MikhailAMD** [comment]
I installed the latest public driver - the same as you are 17.4.1. But I uninstalled all AMD drivers and even deleted AMD folders in driverstore. Do you know how to do this?

**Comment by Xaymar** [comment]
Yes, I know how to do it. I had to do it before when I encountered another issue - even have it documented for the plugin. I've redownloaded the driver now and indeed there are binary differences between the one I downloaded now and the one almost 10 hours ago.

**Comment by Xaymar** [comment]
I've installed a completely clean 17.4.1 now, I still get the same result. Immediately crashes using the VCEEncoderD3D sample tool and the OBS plugin, for the OBS plugin it's on InitOpenCL():

- [VCEEncoderD3D Dump](http://cdn.xaymar.com/private/2017/04/05/17/VCEEncoderD3D.7z)
- [OBS Dump](http://cdn.xaymar.com/private/2017/04/05/17/obs64.7z)

**Comment by MikhailAMD** [comment]
Here is the problem: if in VS2015 you will check Modules View you will find following line:
amdocl12cl64.dll	C:\Program Files (x86)\AMD APP SDK\3.0\bin\x86_64\amdocl12cl64.dll	N/A	Yes	Cannot find or open the PDB file.		62	0, 8, 0, 0, 326	8/4/2015 11:43 PM

The DLL is loaded from APP SDK and not from the driver. Just delete the bin SDK folder or remove it from global path. I will open ticket to OpenCL guys.

**Comment by Xaymar** [comment]
I've removed the APP SDK from PATH now, it seems to load the corect OpenCL binary now. It isn't crashing anymore, so that was probably what caused it. It's odd that the new driver looks for the APP SDK one first instead of the one right next to amfrt*.dll. Thanks for opening a ticket

**Comment by SPIKEYPUP** [comment]
I can confirm this issue as well with 17.4.1 Relive Drivers, upon advice of Mikhail I also wiped out my SDK path and cleared the variables, OBS no longer fails to start, prior this bug was preventing OBS 18.0.1 with 1.9.9.6 AMF Encoder Plugin from fully starting.  Thanks for figuring this out, it was driving me nuts, I thought my VS install was an issue since the installer had flubbed an install the same day I changed to 17.4.1, so there went 28GB.  

_### **Sidenote:** I bring up the Visual Studio incident because I recommend waiting on doing any VS installer activity if you can avoid it for the next few days, MS had a few major bugs in the installer/updater that rendered MANY installations corrupt or dead, forcing reinstall, but snag there...since so many installs have been corrupted, so many people are D/L VS again, and their servers are clogged beyond belief._

**Comment by MikhailAMD** [comment]
should be fixed in 17.4.4 release from today.

**Comment by Xaymar** [comment]
Unable to reproduce on 17.4.4, so I'm guessing it is fixed now.
